src/Testing/Mocks/Tenant.php

```diff
@@ -31,4 +31,27 @@ class Tenant extends Model implements Contract
     protected $table = 'users';
 
     use AllowsTenantIdentification;
+
+    public static function factory(...$parameters)
+    {
+        if (function_exists('factory')) {
+            return factory(get_called_class(), $parameters);
+        }
+
+        return \Illuminate\Database\Eloquent\Factories\Factory::factoryForModel(get_called_class())
+                    ->count(is_numeric($parameters[0] ?? null) ? $parameters[0] : null)
+                    ->state(is_array($parameters[0] ?? null) ? $parameters[0] : ($parameters[1] ?? []));
+    }
```

Comment:
Basically:
- Support old factory usage
- Support new factory usage, by copy pasting the hasFactory trait :)